pkg/k8s/kubevirt.go

```diff
@@ -0,0 +1,348 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	b64 "encoding/base64"
+
+	"github.com/cloud-bulldozer/k8s-netperf/pkg/config"
+	kubevirtv1 "github.com/cloud-bulldozer/k8s-netperf/pkg/kubevirt/client-go/clientset/versioned/typed/core/v1"
+	log "github.com/cloud-bulldozer/k8s-netperf/pkg/logging"
+	"github.com/melbahja/goph"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	v1 "kubevirt.io/api/core/v1"
+)
+
+var (
+	sshPort = uint(32022)
+	retry   = 30
+)
+
+func connect(config *goph.Config) (*goph.Client, error) {
+	for i := 0; i < retry; i++ {
+		client, err := goph.NewConn(config)
+		if err != nil {
+			log.Debug("Waiting for ssh access to be available")
+			log.Debug(err)
+			time.Sleep(10 * time.Second)
+			continue
+		} else {
+			return client, nil
+		}
+	}
+	return nil, fmt.Errorf("Unable to connect via ssh after %d attempts", retry)
+}
+
+func SSHConnect(conf *config.PerfScenarios) (*goph.Client, error) {
+	dir, err := os.UserHomeDir()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve users homedir. %s", err)
+	}
+	key := fmt.Sprintf("%s/.ssh/id_rsa", dir)
+	keyd, err := os.ReadFile(key)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to read key. Error : %s", err)
+	}
+	auth, err := goph.RawKey(string(keyd), "")
+	if err != nil {
+		return nil, fmt.Errorf("Unable to retrieve sshkey. Error : %s", err)
+	}
+	user := "fedora"
+	addr := conf.VMHost
+	log.Debugf("Attempting to connect with : %s@%s", user, addr)
+
+	config := goph.Config{
+		User:     user,
+		Addr:     addr,
+		Port:     sshPort,
+		Auth:     auth,
+		Callback: ssh.InsecureIgnoreHostKey(),
+	}
+
+	client, err := connect(&config)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to connect via ssh. Error: %s", err)
+	}
+
+	return client, nil
+}
+
+func createCommService(client *kubernetes.Clientset, label map[string]string, name string) error {
+	log.Infof("🚀 Creating service for %s in namespace %s", name, namespace)
+	sc := client.CoreV1().Services(namespace)
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Ports: []corev1.ServicePort{
+				{
+					Name:       name,
+					Protocol:   corev1.ProtocolTCP,
+					NodePort:   int32(sshPort),
+					TargetPort: intstr.Parse(fmt.Sprintf("%d", 22)),
+					Port:       22,
+				},
+			},
+			Type:     corev1.ServiceType("NodePort"),
```

Comment:
It's not needed a NodePort service

```suggestion
			Type:     corev1.ServiceType("ClusterIP"),
```